### PR TITLE
SQL: Add basic support for geo point

### PR DESCRIPTION
**Pull request by imotov**
Adds basic support for geo point type. For now, the geopoint is
represented as a string and returned in the same format it was stored
in the source.

Relates to #29872

### CODE REVIEW

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysParserTests.java [hunk 1]
```diff
@@ -58,11 +58,11 @@ public void testSysTypes() throws Exception {
         Command cmd = sql("SYS TYPES").v1();
 
         List<String> names = asList("BYTE", "LONG", "BINARY", "NULL", "INTEGER", "SHORT", "HALF_FLOAT", "SCALED_FLOAT", "FLOAT", "DOUBLE",
-                "KEYWORD", "TEXT", "BOOLEAN", "DATE", "UNSUPPORTED", "GEO_SHAPE", "OBJECT", "NESTED");
+                "KEYWORD", "TEXT", "BOOLEAN", "DATE", "UNSUPPORTED", "GEO_SHAPE", "GEO_POINT", "OBJECT", "NESTED");
 
         cmd.execute(null, ActionListener.wrap(r -> {
             assertEquals(19, r.columnCount());
-            assertEquals(18, r.size());
+            assertEquals(19, r.size());
```
Comment (nik9000) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysParserTests.java]: `assertEquals(names.size, r.size());`?

**Review by nik9000 — APPROVED** [review]
I think this just allows you to return and describe geo points without doing anything interesting with them, right?

**Comment by imotov** [comment]
@nik9000 correct, just trying to keep PRs manageable.

**Review by costin — APPROVED** [review]
LGTM

**Pull request merged**